core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1941,11 +1941,19 @@ class KafkaApis(val requestChannel: RequestChannel,
     // with client authentication which is performed at an earlier stage of the connection where the
     // ApiVersionRequest is not available.
     val apiVersionRequest = request.body[ApiVersionsRequest]
+    val expectedClientSoftwareNames = config.expectedClientSoftwareNames
```

Comment:
Do you know if there's anywhere "epectedClientSoftwareNames" gets logged? My biggest concern here is that the black box between setting the config in Xinfra and it showing up here doesn't behave as expected or other values find there way in here which aren't expected that will mess up this logic.

Reply:
not sure, you can check broker logs to see if KafkaConfig is printed anywhere, I think it should be printed somehow; but if not, I think it's good to print all the configs broker used